### DISCUSSION TITLE
Don't store conversions that make no difference

### DISCUSSION THREAD

**Discussion by Synchro** [spatie/laravel-medialibrary#2200] [Ideas] [answered]
Given a conversion like this:

        $this->addMediaConversion('maxsize')
            ->fit(Manipulations::FIT_MAX, 1024, 1024);

if an image is below this size limit, nothing needs doing, however, ML saves a new version anyway. A nice optimisation would be to somehow detect null operations like this, and serve the original image instead, saving both storage space and processing time.

The hard part is detecting this! I guess ideally feedback from the manipulation would say whether it did anything or not. An alternative might be to process it, then take a hash of the result and see if it's the same as the original. That would still incur processing time, but would still save io bandwidth and storage space.

**Comment by freekmurze** [comment, marked as answer]
I think that total processing power needed for detecting this will far outweigh the benefits for that one upload that might be identical. I also think storage space is cheap. 

So most users won't need an optimisation like this and I prefer to keep to code very simple. 

I might change my stance on this if this would be a problem for more users.

Thanks for suggestion this though.
